### DISCUSSION TITLE
How to Link and Preload FontAwesome Pro Icons in the Head of eXo Platform?

### DISCUSSION THREAD

**Discussion by aimanskie** [exoplatform/exo-community#48] [Q&A]
Hello eXo Community,

I am trying to integrate FontAwesome Pro icons into our eXo Platform. I noticed that the platform preloads fonts in the `<head>`. FontAwesome provides a CDN, and I’m considering using it for ease of integration.

Could you guide me on how to correctly link and preload the FontAwesome Pro icons in the `<head>` of the platform, similar to the existing fonts?

If using the FontAwesome CDN is not recommended or not possible, could you suggest an alternative way to load our FontAwesome Pro icons efficiently?

Any guidance or best practices shared would be greatly appreciated!

Thank you!

**Comment by rdenarie** [comment]
Hello @aimanskie 

In function of your situation, there is a lot of possibilities : 

**Case 1** 
You have develop a portlet for eXo, in which you want to use fontawesone.
So, in the file portlet.xml, you can add a preload resource. Example : https://github.com/Meeds-io/social/blob/develop/webapp/portlet/src/main/webapp/WEB-INF/portlet.xml#L409
Then, when the portlet is displayed, <link with preload will be added in <head>

**Case 2** 
If you already have an extension, but without a portlet, or if you want to have this font for all pages, you can create a [gtmpl,](https://github.com/Meeds-io/social/blob/develop/webapp/portlet/src/main/webapp/groovy/social/webui/UISocialPreloadHead.gtmpl) and add configuration like this : https://github.com/Meeds-io/social/blob/develop/extension/war/src/main/webapp/WEB-INF/conf/social-extension/social/webui-configuration.xml#L52-L66
The gtmpl will be included in all pages, and if you add a link preload into, you will have it every where


**Case 3** 
If you don't have an extension, or don't want to do it by configuration, you can create a simple js file in back office, which will be loaded in all pages. This file will get and load the css or script you want. It will be not preloaded.
For that, go to "Administration", "Back office", and then navigate to folder shared/js. Create a new js content and add this as content : 

`document.getElementsByTagName('head')[0].insertAdjacentHTML(
    'beforeend',
    '<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css" />');`

Adapt the css url, and the code if you need to use a js file.

**Case 4** 
Previous cases can be adapted if you prefer to have the local file file instead a distant one.

**Reply by aimanskie** [reply]
Hi @rdenarie, 

Thank you so much, the icons are showing! In my case, i'm using case 3 and It worked! I truly appreciate the time and effort you took to help.

Aiman

**Reply by rdenarie** [reply]
You're welcome :)
Which case do you use ?

**Reply by aimanskie** [reply]
I used case 3 ![image](https://github.com/exoplatform/exo-community/assets/86446622/1840ae25-5841-4365-8286-853c37056f06)

**Reply by rdenarie** [reply]
Thanks for your answer :)
